ext/odbc/php_odbc.c

```diff
@@ -190,28 +196,21 @@ static void odbc_result_free(odbc_result *res)
 		efree(res->param_info);
 		res->param_info = NULL;
 	}
+
+	HashTable *tmp = res->conn_ptr->results;
 	res->conn_ptr = NULL;
+	if (tmp) {
+		zend_hash_index_del(tmp, res->index);
```

Comment:
When a result is closed, it gets removed from the connection list without the need to iterate the list over.